cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -923,6 +927,142 @@ public StopProposalResult stopProposalExecution(StopProposalParameters parameter
     return new StopProposalResult(_config);
   }
 
+  private void ensureNoOfflineReplicaForPartition(PartitionInfo partitionInfo, List<Node> aliveNodes) {
+    for (Node node: partitionInfo.replicas()) {
+      if (!aliveNodes.contains(node)) {
+        throw new RuntimeException(String.format("Topic partition %s-%d have offline replica on broker %d , unable to update "
+                                   + "its replication factor.", partitionInfo.topic(), partitionInfo.partition(), node.id()));
+      }
+    }
+  }
+
+  /**
+   * Update the replication factor of Kafka topics.
+   * If partition's current replication factor is less than target replication factor, add new replicas to the partition
+   * in a rack-aware, round-robin way.
+   * There are two scenarios that rack awareness property is not guaranteed.
+   * <ul>
+   *   <li> If metadata does not have rack information about brokers, then it is only guaranteed that new replicas are
+   *   added to brokers which currently do not host any replicas of partition.</li>
```

Comment:
Nit: missing comma before `which`.

Reply:
This is definitely not critical -- just fyi, but feel free to skip this: https://www.grammarly.com/blog/comma-before-which/